polkadot/runtime/parachains/src/paras_inherent/tests.rs

```diff
@@ -943,6 +943,57 @@ mod enter {
 		});
 	}
 
+	// Ensure that even if the block is over weight due to candidates enactment,
+	// we still can import it.
+	#[test]
+	fn overweight_candidates_enactment_is_fine() {
+		sp_tracing::try_init_simple();
+		new_test_ext(MockGenesisConfig::default()).execute_with(|| {
+			let mut backed_and_concluding = BTreeMap::new();
+			// The number of candidates is chosen to go over the weight limit
+			// of the mock runtime together with the `enact_candidate`s weight.
```

Comment:
can we assert this?

Reply:
See [0f05cdc](https://github.com/paritytech/polkadot-sdk/pull/5270/commits/0f05cdce33e512c6e339d087cdc911ee8b77c608)